### DISCUSSION TITLE
Information about the lineage reconstruction algorithm

### DISCUSSION THREAD

**Discussion by NoELJ22** [milaboratory/mixcr#1350] [Q&A] [answered]
Hello ,

I would like to have more information about the lineage reconstruction algorithm. 
I understand that it groups clones with the same V and J genes, but does it also use the CDR3 size and homology between CDR3s?
Also, when constructing the links between clones, does the algorithm use the full-length clonotype sequence or just the V sequence?

Thank you for your help.

**Comment by mizraelson** [comment, marked as answer]
The SHM trees algorithm operates on the complete length of the receptor sequence (including CDR3). Below is a concise breakdown of the process:

1. Categorize all clonotypes based on identical V gene alleles and J gene alleles.
2. Within these categories, group clonotypes that share a minimum of 5 mutual mutations and possess a similar NDN-region. Prioritize groupings that encompass the most clonotypes. Use maximum parsimony to construct initial trees.
3. Incorporate other clonotypes that can be added to the trees without altering their structure, ensuring they meet the NDN-region similarity threshold.
4. Determine the most recent common ancestors (MRCAs). Subsequently, combine trees with MRCAs that fall within a set distance threshold, considering both the NDN-region and similarities in the V- and J-genes. 

This method ensures a structured and systematic approach to analyzing receptor sequences and their evolutionary trajectories.

**Comment by NoELJ22** [comment]
Hello,
Thank you very much for your clear answer.
